controller/GiPlanned.controller.js

```diff
@@ -0,0 +1,408 @@
+], function(Controller, MessageToast, JSONModel, MessageBox) {
+	"use strict";
+
+	var GlobalFuncs = sap.ui.controller("smud.org.wmLM01.controller.GlobalFuncs");
+	var zController;
+
+	return Controller.extend("smud.org.wmLM01.controller.GiPlanned", {
+
+		/**
+		 * Called when a controller is instantiated and its View controls (if available) are already created.
+		 * Can be used to modify the View before it is displayed, to bind event handlers and do other one-time initialization.
+		 * @memberOf smud.org.wmLM01.view.GiPlanned
+		 */
+		onInit: function() {
+			zController = this;
+			this.getView().setModel(sap.ui.getCore().getModel("GiPlanned"));
+			this.createTable();
+
+			this._oView = this.getView();
+			this._oView.addEventDelegate({
+				onBeforeHide: function(oEvent) {
+					// Remove items from the table and clear filter search on BACK
+					sap.ui.getCore().byId(zController.createId("materialTable")).removeAllItems();
+					sap.ui.getCore().byId(zController.getView().createId("materialTableSearch")).setValue("");
+					zController.getView().getModel().setProperty("/Tanum", "");
+					zController.getView().getModel().setProperty("/Benum", "");
+				},
+				onAfterHide: function(oEvent) {
+					//     debugger;
+				}
+			}, this);
+		},
+
+		/**
+		 * Similar to onAfterRendering, but this hook is invoked before the controller's View is re-rendered
+		 * (NOT before the first rendering! onInit() is used for that one!).
+		 * @memberOf smud.org.wmLM01.view.GiPlanned
+		 */
+		//	onBeforeRendering: function() {
+		//
+		//	},
+
+		/**
+		 * Called when the View has been rendered (so its HTML is part of the document). Post-rendering manipulations of the HTML could be done here.
+		 * This hook is the same one that SAPUI5 controls get after being rendered.
+		 * @memberOf smud.org.wmLM01.view.GiPlanned
+		 */
+		//	onAfterRendering: function() {
+		//
+		//	},
+
+		/**
+		 * Called when the Controller is destroyed. Use this one to free resources and finalize activities.
+		 * @memberOf smud.org.wmLM01.view.GiPlanned
+		 */
+		//onExit: function() {
+
+		//},
+		getItems: function() {
+			var oTable = sap.ui.getCore().byId(zController.getView().createId("materialTable"));
+			var oModel = zController.getView().getModel();
+			var data = zController.getView().getModel().getData();
+			oTable.removeAllItems();
+			if (oTable.getBinding("items")) {
+				oTable.getBinding("items").refresh();
+			}
+			oModel.setProperty("/Benum", "");
+			if (!data.Tanum) {
+				MessageBox.show("Enter a Transfer Order", {
+					title: "Error",
+					icon: "ERROR"
+				});
+				return;
+			}
+			// Busy dialog...
+			var busy = new sap.m.BusyDialog({
+				text: "Retrieving transfer order items to confirm"
+			});
+			busy.open();
+
+			// Filters for Entity Set
+			var filters = [
+				new sap.ui.model.Filter({
+					path: "Lgnum",
+					operator: sap.ui.model.FilterOperator.EQ,
+					value1: data.Lgnum
+				}),
+				new sap.ui.model.Filter({

```

Comment:
If you use sap.ui.model.Filter a lot of times, you can import it as a module and you don't need to include the full path each time. same for sap.ui.model.FilterOperator.EQ